### PR TITLE
Feature/1109

### PR DESCRIPTION
**Pull request by fraincs**
Issue: 

## Summary

Icons needed a whole redesign in order to look similar when next to each other like in a navigation context. They have been redesigned as 24x24 icons, removing the need for 32x32 icons. In the rare cases we need a bigger icon the 24x24 icon will be resized to 32x32, these have been tested to scale up well.

## What I did

- Replaced the icons with the newly redesigned icons, as well as adding new icons.
- Reworked the naming convention for the icons and the icon files.
- Reworked the doc.
- Removed the multiVariant icon component as all icons will be 24x24, no variants exists for icons anymore.

Note
#1091 depends on this story, in order to avoid a huge PR, although this one is huge already, this won't be merged until #1091 is done.

### CODE REVIEW

File: packages/components/src/autocomplete/tests/chromatic/Autocomplete.chroma.jsx [hunk 1]
```diff
@@ -0,0 +1,482 @@
+import { Autocomplete } from "@components/autocomplete";
```
Comment (alexasselin008) [inline packages/components/src/autocomplete/tests/chromatic/Autocomplete.chroma.jsx:1]: This file should not be added. Its been renamed to .stories.tsx and this file added is the result of a merge conflict gone wrong

File: packages/components/src/badge/tests/chromatic/BadgeIcon.chroma.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import { PlaceholderMajorIcon } from "@components/icons";
```
Comment (alexasselin008) [inline packages/components/src/badge/tests/chromatic/BadgeIcon.chroma.jsx:1]: This file should not be added. Its been renamed to .stories.tsx and this file added is the result of a merge conflict gone wrong

File: packages/components/src/button/tests/chromatic/ButtonGroup.chroma.jsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
+import { AddMajorIcon } from "@components/icons";
```
Comment (alexasselin008) [inline packages/components/src/button/tests/chromatic/ButtonGroup.chroma.jsx:1]: This file should not be added. Its been renamed to .stories.tsx and this file added is the result of a merge conflict gone wrong

File: packages/components/src/field/tests/chromatic/FieldMessage.chroma.jsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
+import { Div, LI, OL, UL } from "@components/html";
```
Comment (alexasselin008) [inline packages/components/src/field/tests/chromatic/FieldMessage.chroma.jsx:1]: This file should not be added. Its been renamed to .stories.tsx and this file added is the result of a merge conflict gone wrong

File: packages/components/src/icons/docs/IndexFileUsage.mdx [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
-    import { ReactComponent as Azure32 } from "./icon-azure-32.svg";
-    import { ReactComponent as Filter24 } from "./icon-filter-24.svg";
-    import { ReactComponent as Filter32 } from "./icon-filter-32.svg";
+    import { createIcon } from "@sharegate/orbit-ui";
+    import { ReactComponent as Azure } from "./icon-azure.svg";
+    import { ReactComponent as Filter } from "./icon-filter.svg";
 
-    export const AzureIcon32 = createIcon(Azure32);
-    export const FilterIcon = createMultiVariantIcon(Filter24, Filter32);
+    export const AzureIcon = createIcon(Azure32);
```
Comment (alexasselin008) [inline packages/components/src/icons/docs/IndexFileUsage.mdx:11]: pas besoin de spcéfié 32

Comment (alexasselin008) [inline packages/components/src/icons/docs/IndexFileUsage.mdx:11]: should add Major/Minor to the icon name

File: packages/components/src/icons/src/Icon.tsx [hunk 6]
```diff
@@ -4,7 +4,7 @@ import { ResponsiveProp, StyledSystemProps, useResponsiveValue, useStyleProps }
 
 import { Box } from "../../box";
 
-export type IconSize = "2xs" | "xs" | "sm" | "md" | "lg" | "xl" | "inherit";
+export type IconSize = "inherit";
```
Comment (alexasselin008) [inline packages/components/src/icons/src/Icon.tsx:7]: Should we allow "md" and inherit, or only inherit? not sure if its weird that the only allowed size is inherit

File: packages/components/src/input-group/tests/chromatic/InputGroup.chroma.jsx [hunk 7]
```diff
@@ -0,0 +1,975 @@
+import { Autocomplete } from "@components/autocomplete";
+import { Button, IconButton } from "@components/button";
+import { DateInput, DateRangeInput } from "@components/date-input";
```
Comment (alexasselin008) [inline packages/components/src/input-group/tests/chromatic/InputGroup.chroma.jsx:3]: This file should not be added. Its been renamed to .stories.tsx and this file added is the result of a merge conflict gone wrong

File: packages/components/src/listbox/tests/chromatic/Listbox.chroma.jsx [hunk 8]
```diff
@@ -0,0 +1,672 @@
+import { Avatar } from "@components/avatar";
+import { PlaceholderMajorIcon, CrossMajorIcon, IconList, LightbulbMajorIcon, NotificationMajorIcon } from "@components/icons";
```
Comment (alexasselin008) [inline packages/components/src/listbox/tests/chromatic/Listbox.chroma.jsx:2]: This file should not be added. Its been renamed to .stories.tsx and this file added is the result of a merge conflict gone wrong

File: packages/components/src/lozenge/tests/chromatic/Lozenge.chroma.jsx [hunk 9]
```diff
@@ -0,0 +1,218 @@
+import { PlaceholderMajorIcon, IconList } from "@components/icons";
```
Comment (alexasselin008) [inline packages/components/src/lozenge/tests/chromatic/Lozenge.chroma.jsx:1]: This file should not be added. Its been renamed to .stories.tsx and this file added is the result of a merge conflict gone wrong

File: packages/components/src/menu/tests/chromatic/Menu.chroma.jsx [hunk 10]
```diff
@@ -0,0 +1,747 @@
+import { Avatar } from "@components/avatar";
```
Comment (alexasselin008) [inline packages/components/src/menu/tests/chromatic/Menu.chroma.jsx:1]: This file should not be added. Its been renamed to .stories.tsx and this file added is the result of a merge conflict gone wrong

File: packages/components/src/menu/tests/chromatic/MenuTrigger.chroma.jsx [hunk 11]
```diff
@@ -0,0 +1,296 @@
+import { Button, IconButton } from "@components/button";
+import { Item, Section } from "@components/collection";
+import { Menu, MenuTrigger } from "@components/menu";
+import { DisclosureArrow } from "@components/disclosure";
+import { Divider } from "@components/divider";
+import { HtmlButton } from "@components/html";
```
Comment (alexasselin008) [inline packages/components/src/menu/tests/chromatic/MenuTrigger.chroma.jsx:6]: This file should not be added. Its been renamed to .stories.tsx and this file added is the result of a merge conflict gone wrong

File: packages/components/src/number-input/src/NumberInput.css [hunk 12]
```diff
@@ -71,6 +71,9 @@
 
 .o-ui-number-input-spinner .o-ui-icon {
     fill: var(--o-ui-icon-alias-primary);
+    align-self: center;
+    width: 1rem;
```
Comment (alexasselin008) [inline packages/components/src/number-input/src/NumberInput.css:75]: all icons inside a number input are reduced in size to 16x16?

Comment (alexasselin008) [reply packages/components/src/number-input/src/NumberInput.css:75]: looks like a mistake

File: packages/components/src/styling/tests/chromatic/sharegate.chroma.jsx [hunk 13]
```diff
@@ -0,0 +1,145 @@
+import {
+    BackgroundColorMapping,
```
Comment (alexasselin008) [inline packages/components/src/styling/tests/chromatic/sharegate.chroma.jsx:2]: This file should not be added. Its been renamed to .stories.tsx and this file added is the result of a merge conflict gone wrong

File: packages/components/src/tabs/tests/chromatic/Tabs.chroma.jsx [hunk 14]
```diff
@@ -0,0 +1,686 @@
+import { PlaceholderMajorIcon, CrossMajorIcon, NotificationMajorIcon } from "@components/icons";
```
Comment (alexasselin008) [inline packages/components/src/tabs/tests/chromatic/Tabs.chroma.jsx:1]: This file should not be added. Its been renamed to .stories.tsx and this file added is the result of a merge conflict gone wrong

File: packages/components/src/tabs/tests/chromatic/TabsVertical.chroma.jsx [hunk 15]
```diff
@@ -0,0 +1,181 @@
+import { NotificationMajorIcon } from "@components/icons";
```
Comment (alexasselin008) [inline packages/components/src/tabs/tests/chromatic/TabsVertical.chroma.jsx:1]: This file should not be added. Its been renamed to .stories.tsx and this file added is the result of a merge conflict gone wrong

File: packages/components/src/tag/tests/chromatic/TagList.chroma.jsx [hunk 16]
```diff
@@ -0,0 +1,189 @@
+import { Counter } from "@components/counter";
```
Comment (alexasselin008) [inline packages/components/src/tag/tests/chromatic/TagList.chroma.jsx:1]: This file should not be added. Its been renamed to .stories.tsx and this file added is the result of a merge conflict gone wrong

File: packages/components/src/text-input/tests/chromatic/SearchInput.chroma.jsx [hunk 17]
```diff
@@ -0,0 +1,144 @@
+import { Inline, Stack } from "@components/layout";
```
Comment (alexasselin008) [inline packages/components/src/text-input/tests/chromatic/SearchInput.chroma.jsx:1]: This file should not be added. Its been renamed to .stories.tsx and this file added is the result of a merge conflict gone wrong

File: packages/components/src/text-input/tests/chromatic/TextInput.chroma.jsx [hunk 18]
```diff
@@ -0,0 +1,160 @@
+import { Inline, Stack } from "@components/layout";
+
+import { CrossButton } from "@components/button";
```
Comment (alexasselin008) [inline packages/components/src/text-input/tests/chromatic/TextInput.chroma.jsx:3]: This file should not be added. Its been renamed to .stories.tsx and this file added is the result of a merge conflict gone wrong

File: packages/components/src/icons/src/Icon.tsx (hunk 6 above)
Comment (fraincs) [reply packages/components/src/icons/src/Icon.tsx:7]: md would be placebo as anything not inherit would be md

File: packages/components/src/number-input/src/NumberInput.css (hunk 12 above)
Comment (fraincs) [reply packages/components/src/number-input/src/NumberInput.css:75]: All icons in an input spinner which is the buttons to up down the numbers

Comment (alexasselin008) [reply packages/components/src/number-input/src/NumberInput.css:75]: you're right, my bad

File: packages/experimental/src/contextual-help/src/ContextualHelp.tsx [hunk 19]
```diff
@@ -1,17 +1,17 @@
...
+export interface InnerContextualHelpProps extends ComponentProps<typeof InfoCircleMajorIcon>, SlotProps, InternalProps{
     children: ReactNode;
     tooltipProps?: Partial<TooltipProps>;
     tooltipTriggerProps?: Partial<TooltipTriggerProps>;
     variant?: "help" | "info";
 }
 
 export function InnerContextualHelp({ children, forwardedRef, tooltipProps, tooltipTriggerProps, variant, ...rest }: InnerContextualHelpProps) {
-    const Icon = variant === "help" ? HelpIcon : InfoIcon;
+    const Icon = variant === "help" ? HelpMajorIcon : InfoCircleMajorIcon;
```
Comment (alexasselin008) [inline packages/experimental/src/contextual-help/src/ContextualHelp.tsx:14]: HelpCircleMajorIcon

File: packages/components/src/icons/src/Icon.tsx (hunk 6 above)
Comment (fraincs) [reply packages/components/src/icons/src/Icon.tsx:7]: @alexasselin008

**Comment by alexasselin008** [comment]
image

mask-size isnt a valid property?

**Comment by alexasselin008** [comment]
Checkbox small's checkmark is really blurry 
image

**Comment by alexasselin008** [comment]
The tile's checkmark is reduced to 16x16, but i beleive it should be 20x20. There is no reason to reduce the size in this occurence

```
.o-ui-tile-main::after {
    width: var(--o-ui-sz-2);
    height: var(--o-ui-sz-2);
```

**Comment by alexasselin008** [comment]
image
looks a bit blurry at the bottom

**Comment by fraincs** [comment]
> 

will investigate a solution

**Comment by fraincs** [comment]
> image looks a bit blurry at the bottom

will need an opinion from design it indeed does but was flagged as OK.

**Comment by fraincs** [comment]
> The tile's checkmark is reduced to 16x16, but i beleive it should be 20x20. There is no reason to reduce the size in this occurence
> 
> ```
> .o-ui-tile-main::after {
>     width: var(--o-ui-sz-2);
>     height: var(--o-ui-sz-2);
> ```

It is resized to 16x16 in a 20x20 circle, emulating a CheckCircle icon that we don't have anymore, a 20x20 icon.

**Comment by fraincs** [comment]
> 

Might be wrongly configured? Something to check.

**Review by alexasselin008 — APPROVED** [review]

**Pull request merged**